File: Net/Map/AddInReportManager/VBNET/GenerateReport.vb
Public Class GenerateReport
  Inherits ESRI.ArcGIS.Desktop.AddIns.Button

  Public Sub New()

  End Sub

  Protected Overrides Sub OnClick()
        My.ArcMap.Application.CurrentTool = Nothing
        Dim exportReport As ExportReport = New ExportReport()
        exportReport.Show()
  End Sub

  Protected Overrides Sub OnUpdate()
    Enabled = My.ArcMap.Application IsNot Nothing
  End Sub
End Class

File: Net/Map/AddInReportManager/CSharp/GenerateReport.cs
using ESRI.ArcGIS.Framework;
using ESRI.ArcGIS.esriSystem;
using ESRI.ArcGIS.ArcMapUI;
using ESRI.ArcGIS.ArcMap;
using ESRI.ArcGIS.Carto;
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace CustomReport_CS
{
    public class GenerateReport : ESRI.ArcGIS.Desktop.AddIns.Button
    {
        public GenerateReport()
        {
        }

        protected override void OnClick()
        {
            ArcMap.Application.CurrentTool = null;
            ExportReport exportReport = new ExportReport();
            exportReport.Show();
        }
        protected override void OnUpdate()
        {
            Enabled = ArcMap.Application != null;
        }       
    }

}
